smoke-test.go

```diff
@@ -0,0 +1,279 @@
+	if err != nil {
+		fmt.Printf("Error getting notification methods to delete potential left over notification method "+
+			"from previous runs - %s\n", err.Error())
+	} else {
+		notificationID := ""
+		for _, notificationMethod := range notificationMethods.Elements {
+			if notificationMethod.Name == notificationName {
+				notificationID = notificationMethod.ID
+			}
+		}
+		if notificationID != "" {
+			err := monascaclient.DeleteNotificationMethod(notificationID)
+			if err != nil {
+				fmt.Printf("Error deleting notification method - %s\n", err.Error())
+			}
+		}
+	}
+	alarmDefinitionName := "smoke_test_alarm"
+	alarmDefinitions, err := monascaclient.GetAlarmDefinitions(&models.AlarmDefinitionQuery{Name: &alarmDefinitionName})
+	if err != nil {
+		fmt.Printf("Error getting alarm definitions to delete potential left over alarm definition "+
+			"from previous runs - %s\n", err.Error())
+	} else {
+		alarmDefinitionID := ""
+		for _, alarmDefinition := range alarmDefinitions.Elements {
+			if alarmDefinition.Name == alarmDefinitionName {
+				alarmDefinitionID = alarmDefinition.ID
+			}
+		}
+		if alarmDefinitionID != "" {
+			err := monascaclient.DeleteAlarmDefinition(alarmDefinitionID)
+			if err != nil {
+				fmt.Printf("Error deleting alarm definition - %s\n", err.Error())
+			}
+		}
+	}
+}
+
+func main() {
+	// Initialize keystone client and get Monasca endpoint
+	keystoneOpts, err := openstack.AuthOptionsFromEnv()
+	testsRun := 0
+
+	if err != nil {
+		fmt.Printf("ERROR setting up keystone client - %s", err.Error())
+		os.Exit(1)
+	}
+
+	monascaURL := os.Getenv("MONASCA_URL")
+
+	if monascaURL == "" {
+		fmt.Println("Monasca URL environment variable must be set")
+		os.Exit(1)
+	}
+
+	keystoneToken, err := getToken(&keystoneOpts)
+
+	if err != nil {
+		fmt.Printf("ERROR getting keystone token - %s", err.Error())
+		os.Exit(1)
+	}
+
+	initializeMonascaClient(keystoneToken, monascaURL)
+
+	// Cleanup potential smoke test alarm definition and notification method from previous runs
+	cleanupPreviousRun()
+
+	fmt.Println("TEST MEASUREMENTS FLOWING")
+	testMeasurementsFlowing()
+	fmt.Println()
+	testsRun++
+
+	// Set Up Webhook server for notifications
+	webhookIP := os.Getenv("WEBHOOK_IP")
+	if webhookIP == "" {
+		webhookIP = "127.0.0.1"
+	}
+	webhookAddress := "http://" + webhookIP + ":8080"
+	http.HandleFunc("/", handleWebhook)
+	go http.ListenAndServe(":8080", nil)
+
+	fmt.Println("TEST NOTIFICATION CREATION")
+	notificationID := testCreateNotification(webhookAddress)
+	fmt.Println()
+	testsRun++
+
+	fmt.Println("TEST ALARM DEFINITION CREATION")
+	alarmDefinitionID := testCreateAlarmDefinition(notificationID)
+	fmt.Println()
+	testsRun++
+
+	fmt.Println("TEST METRIC CREATION")
+	testCreateMetric(1)
+	fmt.Println()
+	testsRun++
+
+	fmt.Println("TEST WEBHOOK TRIGGERED")
+	testWebhookTrigger()
+	fmt.Println()
+	testsRun++
+
+	fmt.Println("TEST CLEANUP")
+	cleanup(alarmDefinitionID, notificationID)
```

Comment:
Personally, I think you should only cleanup if everything else succeeded. Doing it always makes it harder to debug issues.